src/EventListener/SessionExpirationListener.php

```diff
@@ -63,33 +60,37 @@ public function __construct(TokenStorageInterface $tokenStorage, ScopeMatcher $s
      */
     public function onKernelRequest(GetResponseEvent $event): void
     {
-        if (!$this->scopeMatcher->isContaoMasterRequest($event)) {
+        $request = $event->getRequest();
+        $session = $request->getSession();
+
+        if (!$event->getRequest()->hasPreviousSession()
+            || null === $session
+            || !$session->isStarted()
+        ) {
             return;
         }
 
         $token = $this->tokenStorage->getToken();
 
-        if (!$token instanceof UsernamePasswordToken) {
+        if (!$this->scopeMatcher->isContaoMasterRequest($event)
+            || !$token instanceof TokenInterface
+            || !($user = $token->getUser()) instanceof User
+        ) {
             return;
         }
 
-        $user = $token->getUser();
-
-        if (!$user instanceof User) {
+        // Logout after 1 hour
+        if ((time() - $session->getMetadataBag()->getLastUsed()) < 3600) {
```

Comment:
I think this needs to remain configurable. Might be a compliance requirement to reduce the timeout.

Reply:
If we add that configuration, we can't use [Symfony's pull request](https://github.com/symfony/symfony/pull/12807) without dropping the configuration variable again later on… not sure if that is ok?

Reply:
We need to have it configurable, so yes, it is ok to deprecate the variable later.

Of course, we could also name the variable `session_expiration.max_idle_time`, then it would be forward compatible with Symfony 4.1.

Reply:
It would not be forward compatible, because the configuration is per firewall, not a bundle configuration.